django/core/management/color.py

```diff
@@ -14,11 +14,57 @@ def supports_color():
     Return True if the running system's terminal supports color,
     and False otherwise.
     """
-    supported_platform = sys.platform != 'win32' or 'ANSICON' in os.environ
+
+    def supported_platform():
+        """
+        Determines if the current platform will support terminal color.
+        """
+
+        if sys.platform != 'win32':
+            return True
+
+        if 'ANSICON' in os.environ:
+            return True
+
+        # Microsoft Terminal supports VT codes
+        if 'WT_SESSION' in os.environ:
+            return True
+
+        # Microsoft Visual Studio Code's built-in terminal supports colors
+        if 'TERM_PROGRAM' in os.environ and os.environ['TERM_PROGRAM'] == 'vscode':
+            return True
+
+        # check the Windows Registry to see if VT code handling has been
+        # enabled by default; see https://superuser.com/a/1300251/447564
```

Comment:
I'm not sure that it makes any sense checking the registry. There is no guarantee that you are using a terminal that uses this. So it could just result in a blanket "on" if this is set. It would be better to take the `ctypes` approach mentioned to check whether currently enabled for the actual terminal in use.

Reply:
I think the point is that this defaults to `off`, so our chance of a false positive is going to be pretty low no? (In which case folks can just set `nocolors`...) (But, yes, maybe this is a better way...)

The ctypes approach is from Davids https://github.com/django/django/pull/12387#issuecomment-674494510

Reply:
The setting defaults to "off" and so if it's on, you've most likely done it deliberately at some point. I.e. it seems like a deliberate attempt by the user to get color output.

It's possible that a terminal emulation might not pay attention to this setting, but that is only an issue if it also does not process VT colours. I'm not aware of any any that would fall into this (although I also haven't don't an exhaustive survey either...). I.e. This seems more hypothetical than something that we will run into with any frequency.

The `ctypes` approach is probably better, however, it is beyond my (current) Python ability. My fear there is that even if I did get it working, I wouldn't be able to fix it (debug it) when if breaks...